### ISSUE TITLE
Expose GraphSage in C/C++ API

### ISSUE THREAD

**Issue by ChuckHastings** [gunrock/gunrock#824]
Expose the GraphSage algorithm implemented in version 1.x of gunrock through an external API.

**Comment by jowens** [comment]
Just a note that the DARPA GraphSAGE app we developed was inference-only, no training. @wangxiaoyunanne can you give any more details? I know you worked a bit on the training side but I don't believe that work was finished (I think we decided to work on the GraphChallenge submission instead).

**Comment by wangxiaoyunanne** [comment]
Yes, this code is inference only. Because of training we need to have a back propagation function, which we do not have in Gunrock. Another way is to write the weight matrix update function by yourself, you need to derive the gradient formula by yourself. But in this way, there is no flexibility in the model structures.
